pkg/controller/handlers/scheduledauditlogexport/scheduledauditlogexport.go

```diff
@@ -0,0 +1,137 @@
+package scheduledauditlogexport
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/adhocore/gronx"
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/obot-platform/nah/pkg/router"
+	client "github.com/obot-platform/obot/pkg/gateway/client"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Handler struct {
+	client        kclient.Client
+	gptClient     *gptscript.GPTScript
+	gatewayClient *client.Client
+}
+
+func NewHandler(client kclient.Client, gptClient *gptscript.GPTScript, gatewayClient *client.Client) *Handler {
+	return &Handler{
+		client:        client,
+		gptClient:     gptClient,
+		gatewayClient: gatewayClient,
+	}
+}
+
+func (h *Handler) ScheduleExports(req router.Request, resp router.Response) error {
+	scheduledExport := req.Object.(*v1.ScheduledAuditLogExport)
+
+	if !scheduledExport.Spec.Enabled {
+		return nil
+	}
+
+	next, err := calculateNextRunTime(scheduledExport)
+	if err != nil {
+		return fmt.Errorf("failed to calculate next run time: %w", err)
+	}
+
+	if until := time.Until(next); until > 0 {
```

Comment:
```suggestion
	if until := time.Until(next); until > 0 && until < 10 * time.Hour {
```

Reply:
is that because we will requeue every 10 hours regardless?